packages/enzyme/src/Utils.js

```diff
@@ -320,3 +320,49 @@ export function spyMethod(instance, methodName) {
     },
   };
 }
+
+export function validEmptyValues(renderedValue) {
+  return renderedValue === null || renderedValue === false;
+}
+
+export function renderedDive(nodes) {
+  let isEmptyRender = true;
+  const renderedKey = 'rendered';
+
+  function dive(renderedNodes) {
+    const isList = renderedNodes && renderedNodes.length > 0 && Array.isArray(renderedNodes);
+    if (isList) {
+      return renderedNodes.some((n) => {
+        if (n) {
+          if ((has(n, renderedKey))) {
+            dive(n.rendered);
+          } else if (!validEmptyValues(n.rendered)) {
+            isEmptyRender = false;
+          }
+        } else if (!validEmptyValues(n)) {
+          isEmptyRender = false;
+        }
+
+        return !isEmptyRender;
+      });
+    }
+
+    if (!validEmptyValues(renderedNodes)) {
+      if (has(renderedNodes, renderedKey)) {
+        dive(renderedNodes.rendered);
+      } else if (has(renderedNodes, renderedKey) && !validEmptyValues(renderedNodes.rendered)) {
+        isEmptyRender = false;
+      } else if (!has(renderedNodes, renderedKey) && !validEmptyValues(renderedNodes)) {
+        isEmptyRender = false;
+      } else {
+        dive(renderedNodes.rendered);
+      }
+    }
+
+    return isEmptyRender;
+  }
+
+  dive(nodes);
```

Comment:
could this not be `return dive(nodes)`?

Reply:
So initially I did have that.  And then in my utils test cases, I was getting inconsistent return values.  I will change back, bc this is something I prefer and research why.